RO-2422: vis observasjoner fungerer nå, og ble flyttet til filter meny

amish1188 opened the pull request:
Øystein: Har forsøkt å svare ut Jørgens kommentarer og forenklet koden. Nå er det kun HomePage og ObservationListPage som styrer om vi skal vise observasjoner eller ikke.
Nå skjuler vi også resten av filtrene hvis visning av observasjoner er skrudd av, slik det sto i Jira-saken.


_Marcins tidligere beskrivelse er under her, så vi har historikk:_
Vis observasjoner ble ikke egentlig kallet på en riktig måte før, derfor var toggelen menigsløs.

Mitt forslag var å bruke  selectedTab$, i search-registration.service så at vi vet hvilken tab er på, og  om showObservations er på. Ellers så ville kalle Search kalles på home page hvis vi slår vis observasjoner av og så på igjen. Det er ikke en pen løsning med å sammenlikne tabber men det var kjapest jeg kom opp med.

amish1188 committed:
moved show observations toggle to filter menu

amish1188 committed:
added changes

amish1188 committed:
moved that toggle

amish1188 committed:
Merge branch 'develop' of https://github.com/NVE/regObs4 into bug/ro-2422/show-observations-move-to-filters

amish1188 committed:
removed method

jorgkv commented on src/app/pages/home/home.page.ts:
```diff
@@ -178,7 +174,19 @@ export class HomePage extends RouterPage implements OnInit, AfterViewChecked, On
+      this.userSettingService.showObservations$,
+      this.refreshRequested$.pipe(startWith(true)),
+    ]).pipe(
+      filter(([, , showObservations]) => {
```
Hvorfor bruke filter til dette? Filter brukes typisk til å filtrere bort noe fra en liste, feks.

jorgkv replied:
Hvis du endrer denne til en tap, så kan den godt heller ligge direkte på `this.userSettingService.showObservations$`, så den ikke kjøres unødvendig når feks searchCriteria$ endres.

gruble replied:
Valgte å fortsatt bruke filter, men på en annen observable.

jorgkv commented on src/app/core/services/search-registration/search-registration.service.ts:
```diff
@@ -106,11 +109,26 @@ export class PagedSearchResult<TViewModel extends HasRegId> {
     countFunc: (criteria: SearchCriteriaRequestDto) => Observable<number>
   ) {
-    this.registrations$ = combineLatest([searchCriteria$, this.forceUpdate.pipe(startWith(null))]).pipe(
+    console.log('tab from service', tab);
```
Hvorfor console.log her? Vi har jo valgt å ikke tillate bruk av console logging. Enten må du gi loggingService til denne klassen, eller fjerne loggingen hvis det ikke trengs.

gruble replied:
Koden er fjernet

jorgkv commented on src/app/core/services/search-registration/search-registration.service.ts:
```diff
@@ -106,11 +109,26 @@ export class PagedSearchResult<TViewModel extends HasRegId> {
     searchCriteria$: Observable<SearchCriteria>,
     // Not sure what is best here, provide SearchService to this class, or provide the flexibility to create these
     // functions outside
+    tab: string,
```
Jeg syns ikke det blir så ryddig at denne klassen forholder seg til hvilken tab man er på eller om show observations er togglet av eller på. Det er bedre om brukeren av klassen styrer dette med searchCriteria$.

gruble replied:
Enig, fjernet.

jorgkv commented on src/app/modules/side-menu/components/filter-menu/filter-menu.component.scss:
```diff
@@ -20,6 +20,16 @@ ion-list {
   overflow: auto;
   max-height: 100vh;
   background: #fff;
+  ion-item {
```
Som alltid, legg til en kommentar som beskriver hva cssen skal gjøre og hvorfor den trengs.

jorgkv commented on src/app/modules/side-menu/components/side-menu.component.html:
```diff
@@ -59,13 +59,6 @@
     <ion-label class="menu-label"> Vis obsturer </ion-label>
     <ion-toggle [checked]="observerTrips.toggledOn | async" (ionChange)="observerTrips.toggle()"></ion-toggle>
   </ion-item>
-  <ion-item>
```
Siden dette fjernes fra denne templaten, har du sjekket om det er noe css som også kan fjernes?

gruble replied:
Har sjekket nå og fjernet noe som ikke så ut til å gjøre noen forskjell. Men siden vi fortsatt har av/på-knapper som ligner i hovedmenyen, tør jeg ikke fjerne alt.

jorgkv commented on src/app/pages/observation-list/observation-list.page.html:
```diff
@@ -65,17 +65,19 @@ <h3 class="ion-text-uppercase header">{{'OBSERVATION_LIST.TITLE' | translate }}<
-            <ion-infinite-scroll-content> </ion-infinite-scroll-content>
-          </ion-infinite-scroll>
-        </ion-list>
+      <ng-container *ngIf="showObservations$ | async">
```
Jeg tror det blir litt rart å ikke vise observasjoner i listevisninga? Da blir det jo bare en helt tom side. 
Jeg tror vi bør lage en empty state som forklarer at man må skru togglen på for å se noe i lista. Se hvordan empty-staten under er laget.

gruble replied:
Dette fiksa du jo fint Jørgen med din commit. Synes det ble ryddig.

jorgkv commented on src/app/pages/observation-list/observation-list.page.ts:
```diff
@@ -167,11 +171,13 @@ export class ObservationListPage extends NgDestoryBase implements OnInit {
 
     this.listSearch = this.searchRegistrationService.pagedSearch(
-      this.searchCriteriaWhenThisPageIsActiveAndViewTypeList$
+      this.searchCriteriaWhenThisPageIsActiveAndViewTypeList$,
```
Igjen, det blir litt rart at man sender inn hvilken tab man er på til denne metoden. this.searchCriteriaWhenThisPageIsActiveAndViewTypeList$ tar jo allerede høyde for om siden er aktiv eller ikke osv. Hvorfor trengs det at man ser på tab i tillegg?

jorgkv reviewed (changes requested):
Tror dette kan løses litt enklere uten at søke-klassen vet hvilken tab som er aktiv.

jorgkv committed:
Empty state when observations disabled

gruble committed:
Revert "added changes"

This reverts commit 4d870a82fbc7d32d0f83f89278329b03823b6578.

gruble committed:
Viser kun  observasjoner i kart hvis "Vis observasjoner" er på

gruble committed:
Fikset listevisning når observasjoner er skrudd av

noe av koden fra Jørgen gikk tapt i en merge når jeg rullet tilbake Marcins commit

gruble committed:
Kun søke etter observasjoner i kart når kart-fanen er aktiv

gruble committed:
Lar være å søke i listevisning hvis observasjoner er skrudd av

gruble committed:
Skjuler filtre hvis observasjoner er deaktivert

gruble committed:
Fjernet unødvendig css

jorgkv commented on src/app/pages/observation-list/observation-list.page.ts:
```diff
@@ -96,6 +101,12 @@ export class ObservationListPage extends NgDestoryBase implements OnInit {
       )
       .subscribe();
 
+    mapService.mapView$.pipe(
```
Her har det sneket seg med noe kode som ikke gjør noen ting tror jeg?

gruble replied:
Jammen! Det var sløvt at jeg ikke så det.

jorgkv committed:
Remove unneeded css

jorgkv reviewed (approved):
Veldig bra. Se kommentar bare, tror det har sneket seg med noe kode som kan slettes.

gruble committed:
Fjernet kode som kom inn ved en feil

og oppdaget av Jørgen
